### DISCUSSION TITLE
504: GATEWAY_TIMEOUT

### DISCUSSION THREAD

**Discussion by ats1999** [vercel/community#5226] [Feature Question] [answered]
### Question

<img width="1272" alt="Screenshot 2023-12-18 at 9 38 38 PM" src="https://github.com/vercel/community/assets/54087826/54c0d0ec-e33c-4212-a12d-6de31e6071b6">

Tech Stack

1. Next.js (Front End)
2. Spring Boot (Back End)

> It's quite possible that you may not receive the same error because page might have been statically generated.

Here is the page link https://dsabyte.com/blog/dsabyte/Weekly-ByteWise-Insights-0-5716fdd0-bd46-444d-9b8d-bda8b464d467

Just for your information, next.js static generation requires querying Spring Boot backend. Currently, the backend is slow and it may take up to 10s to response to the first request. Is there any way to overcome this without improving backend server?

**Comment by amyegan** [comment, marked as answer]
I can think of a few options for this. The best one for you is going to depend on your project and what you're comfortable doing. 

If you're on a Pro plan, then you can configure a higher [max duration](https://vercel.com/docs/functions/serverless-functions/runtimes#max-duration) value to avoid the [504](https://vercel.com/docs/errors/application-error-codes#504:-function_invocation_timeout) timeout error. 10 seconds is the limit for Hobby plan, but Pro teams can [configure max duration](https://vercel.com/docs/functions/configuring-functions/duration#) for up to 300 seconds (5 minutes).

Another option is to, like you suggested, get the backend responding faster. Sometimes the only thing needed is to switch to a couple of smaller requests instead of returning one large chunk of data.

Depending on how this part of your app is set up, Edge Functions could be a good option. The [maximum initial response time](https://vercel.com/docs/functions/edge-functions/limitations#maximum-initial-response-time) is higher than the hobby plan's Serverless Function duration limit, so an Edge Function needs to start sending a response within 25 seconds.

**Reply by ats1999** [reply]
I am using hobby plan, and edge functions do have more limitations on hobby plan

**Reply by ats1999** [reply]
My bundle size is more than 1.5 mb
